### ISSUE TITLE
What about CLI tools that have a set of commands

### ISSUE THREAD

**Issue by javajon** [pholser/jopt-simple#109]
This API and other CLI apis seem to be very focused on options (switches) and supporting their associated arguments.  However, what about command line tools that have a preset list of documented commands.  Take for instance a command like "minikube".  Minikube has a definitive and enumerated list of commands it can perform.  

minikube [start|stop|status|version|ssh] to name a few. It also has a bunch of options and it also has the notion of "minikube start --help".

I'm not seeing examples on how to support this command line style in this API (I looked at args4j too).  Both CLI APIs appear to focus on the "options" problems, but what about those tools that have a fixed enumeration of commands.  

-----
Here is the help of minikube that I would like to pattern a tool from using the jopt-simple API.  Its this pattern possible with this API?  Is there an example in examples that addresses this pattern?:

Example usage help text.

> minikube
Minikube is a CLI tool that provisions and manages single-node Kubernetes clusters optimized for development workflows.

Usage:
  minikube [command]

Available Commands:
  addons           Modify minikube's kubernetes addons
  config           Modify minikube config
  dashboard        Opens/displays the kubernetes dashboard URL for your local cluster
  delete           Deletes a local kubernetes cluster.
  docker-env       sets up docker env variables; similar to '$(docker-machine env)'
  get-k8s-versions Gets the list of available kubernetes versions available for minikube.
  ip               Retrieve the IP address of the running cluster.
  logs             Gets the logs of the running localkube instance, used for debugging minikube, not user code.
  service          Gets the kubernetes URL for the specified service in your local cluster
  ssh              Log into or run a command on a machine with SSH; similar to 'docker-machine ssh'
  start            Starts a local kubernetes cluster.
  status           Gets the status of a local kubernetes cluster.
  stop             Stops a running local kubernetes cluster.
  version          Print the version of minikube.

Flags:
      --alsologtostderr value         log to standard error as well as files
  -h, --help                          help for minikube
      --log_backtrace_at value        when logging hits line file:N, emit a stack trace (default :0)
      --log_dir value                 If non-empty, write log files in this directory
      --logtostderr value             log to standard error instead of files
      --show-libmachine-logs          Whether or not to show logs from libmachine.
      --stderrthreshold value         logs at or above this threshold go to stderr (default 2)
      --test.bench value              regular expression per path component to select benchmarks to run
      --test.benchmem value           print memory allocations for benchmarks
      --test.benchtime value          approximate run time for each benchmark (default 1s)
      --test.blockprofile value       write a goroutine blocking profile to the named file after execution
      --test.blockprofilerate value   if >= 0, calls runtime.SetBlockProfileRate() (default 1)
      --test.count n                  run tests and benchmarks n times (default 1)
      --test.coverprofile value       write a coverage profile to the named file after execution
      --test.cpu value                comma-separated list of number of CPUs to use for each test
      --test.cpuprofile value         write a cpu profile to the named file during execution
      --test.memprofile value         write a memory profile to the named file after execution
      --test.memprofilerate value     if >=0, sets runtime.MemProfileRate
      --test.outputdir value          directory in which to write profiles
      --test.parallel value           maximum test parallelism (default 8)
      --test.run value                regular expression to select tests and examples to run
      --test.short value              run smaller test suite to save time
      --test.timeout value            if positive, sets an aggregate time limit for all tests (default 0s)
      --test.trace value              write an execution trace to the named file after execution
      --test.v value                  verbose: print additional output
  -v, --v value                       log level for V logs
      --vmodule value                 comma-separated list of pattern=N settings for file-filtered logging

Use "minikube [command] --help" for more information about a command.

**Comment by pholser** [comment]
@javajon Agreed, I'd like to make subcommands simple. I haven't landed on an approach that I like, nor have I been able to spend many cycles on JOpt Simple lately. I'm open to proposals for API to support subcommands.

**Comment by pholser** [comment]
See also #102.
